arcaflow_plugin_fio/fio_schema.py

```diff
@@ -174,128 +228,153 @@ class JobParams:
-            "as described in blocksize."
+            "as described in blocksize. A range is not used by default."
         ),
     ] = None
 
     # I/O  size
     size: typing.Annotated[
         typing.Optional[str],
-        validation.min(2),
-        schema.name("size"),
+        validation.pattern(size_pattern_with_percent),
+        schema.name("Total I/O Size"),
         schema.description(
             "The total size of file I/O for each thread of this job. Fio will run "
-            "until this many bytes has been transferred, unless runtime is altered by "
+            "until this many bytes have been transferred, unless runtime is altered by "
             "other means such as (1) runtime, (2) io_size, (3) number_ios, (4) "
             "gaps/holes while doing I/O's such as 'rw=read:16K', or (5) sequential "
             "I/O reaching end of the file which is possible when percentage_random is "
             "less than 100. Fio will divide this size between the available files "
-            "determined by options such as nrfiles, filename, unless filesize is "
+            "determined by options such as nrfiles or filename, unless filesize is "
             "specified by the job. If the result of division happens to be 0, the size "
             "is set to the physical size of the given files or devices if they exist. "
             "If this option is not specified, fio will use the full size of the given "
             "files or devices. If the files do not exist, size must be given. It is "
             "also  possible to give size as a percentage between 1 and 100. If "
             "'size=20%' is given, fio will use 20% of the full size of the given files "
-            "or devices. In ZBD mode, size can be given in units of number of zones "
-            "using 'z'. Can be combined with offset to constrain the start and end "
-            "range that I/O will be done within."
+            "or devices."
+        ),
+    ] = None
+    io_size: typing.Annotated[
+        typing.Optional[str],
+        validation.pattern(size_pattern_with_percent),
+        schema.name("I/O Size"),
+        schema.description(
+            "Normally fio operates within the region set by size, which means that the "
+            "size option sets both the region and size of I/O to be performed. "
+            "Sometimes that is not what you want. With this option, it is possible to "
+            "define just the amount of I/O that fio should do. For instance, if size "
+            "is set to 20GiB and io_size is set to 5GiB, fio will perform I/O within "
+            "the first 20GiB but exit when 5GiB have been done. The opposite is also "
+            "possible -- if size is set to 20GiB, and io_size is set to 40GiB, then "
+            "fio will do 40GiB of I/O within the 0..20GiB region. Value can be set as "
+            "percentage: io_size=N%. In this case io_size multiplies size= value."
+        ),
```

Comment:
In `io_size=N%`, can `N` be greater than 100?  (If using absolute values, apparently `io_size` can be greater than `size`....)

What is the meaning of the equals sign in line 355?  Should that instead by the word "by"?

Reply:
Weirdly I have a separate problem with this when trying to test for the question you posed. For some reason passing a % value to this parameter is leading to a traceback from `configparser`. I can use a % value with the `size` parameter without issue.

```
output_id: error
output_data:
  error: "Traceback (most recent call last):\n  File \"/app/arcaflow_plugin_fio/fio_plugin.py\"\
    , line 32, in run\n    params.write_jobs_to_file(infile_temp_path)\n  File \"\
    /app/arcaflow_plugin_fio/fio_schema.py\", line 501, in write_jobs_to_file\n  \
    \  cfg[job.name][key] = str(item_value)\n  File \"/usr/lib64/python3.9/configparser.py\"\
    , line 1262, in __setitem__\n    return self._parser.set(self._name, key, value)\n\
    \  File \"/usr/lib64/python3.9/configparser.py\", line 1204, in set\n    super().set(section,\
    \ option, value)\n  File \"/usr/lib64/python3.9/configparser.py\", line 894, in\
    \ set\n    value = self._interpolation.before_set(self, section, option,\n  File\
    \ \"/usr/lib64/python3.9/configparser.py\", line 402, in before_set\n    raise\
    \ ValueError(\"invalid interpolation syntax in %r at \"\nValueError: invalid interpolation\
    \ syntax in '10%' at position 2\n"
```

Reply:
Yeah...`ConfigParser` supports [interpolation](https://docs.python.org/3/library/configparser.html#interpolation-of-values) by [default](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser).  You can either turn it off when you instantiate the parser, or you have to double the `%`'s in the config values to escape them.

Reply:
This doesn't explain why the other case of this with the `size` parameter is working without error ...

Reply:
Turning off interpolation does seem to fix this. Trying to set `io_size` to a value over 100% results in a failure from fio, so we are good with the current pattern.
```
fio: io_size values greater than 100% aren't supported
```

Reply:
> This doesn't explain why the other case of this with the `size` parameter is working without error ...

I _think_ (without actually doing any investigating...) is that what is happening is that you are setting a value of a config option to either (say) `50%` or `2048Kb`; in the latter case, there is no `%`, so there is nothing to interpolate, so the lack of an interpolation value is to be expected and not a problem; but, in the former case, the presence of the unescaped `%` triggers the cascading problem.